types/sarif/index.d.ts

```diff
@@ -0,0 +1,1607 @@
+ * References to external files that should be inlined with the content of a root log file.
+ */
+export interface ExternalFiles {
+    /**
+     * The location of a file containing a run.conversion object to be merged with the root log file.
+     */
+    conversion?: FileLocation;
+
+    /**
+     * The location of a file containing a run.files object to be merged with the root log file.
+     */
+    files?: FileLocation;
+
+    /**
+     * The location of a file containing a run.graphs object to be merged with the root log file.
+     */
+    graphs?: FileLocation;
+
+    /**
+     * An array of locations of files containing arrays of run.invocation objects to be merged with the root log
+     * file.
+     */
+    invocations?: FileLocation[];
+
+    /**
+     * The location of a file containing a run.logicalLocations object to be merged with the root log file.
+     */
+    logicalLocations?: FileLocation;
+
+    /**
+     * The location of a file containing a run.resources object to be merged with the root log file.
+     */
+    resources?: FileLocation;
+
+    /**
+     * An array of locations of files containins arrays of run.result objects to be merged with the root log file.
+     */
+    results?: FileLocation[];
+
+    /**
+     * Key/value pairs that provide additional information about the external files
+     */
+    properties?: PropertyBag;
+}
+
+/**
+ * A single file. In some cases, this file might be nested within another file.
+ */
+export interface File {
+    /**
+     * The contents of the file.
+     */
+    contents?: FileContent;
+
+    /**
+     * Specifies the encoding for a file object that refers to a text file.
+     */
+    encoding?: string;
+
+    /**
+     * The location of the file.
+     */
+    fileLocation?: FileLocation;
+
+    /**
+     * A dictionary, each of whose keys is the name of a hash function and each of whose values is the hashed value
+     * of the file produced by the specified hash function.
+     */
+    hashes?: { [key: string]: string };
+
+    /**
+     * The Coordinated Universal Time (UTC) date and time at which the file was most recently modified. See
+     * "Date/time properties" in the SARIF spec for the required format.
+     */
+    lastModifiedTimeUtc?: string;
+
+    /**
+     * The length of the file in bytes.
+     */
+    length?: number;
+
+    /**
+     * The MIME type (RFC 2045) of the file.
+     */
+    mimeType?: string;
+
+    /**
+     * The offset in bytes of the file within its containing file.
+     */
+    offset?: number;
+
+    /**
+     * Identifies the key of the immediate parent of the file, if this file is nested.
+     */
+    parentKey?: string;
+
+    /**
+     * The role or roles played by the file in the analysis.
+     */
+    roles?: File.roles[];
+
+    /**
+     * Key/value pairs that provide additional information about the file.
+     */
+    properties?: PropertyBag;
+}
+
+export namespace File {
+    // tslint:disable-next-line no-const-enum
```

Comment:
Do you really need to use a const enum? See [here](https://github.com/Microsoft/dtslint/blob/master/docs/no-const-enum.md) for why this rule is enabled.

Reply:
Thanks for the feedback, I've updated these to be "type" instead of "const enum"